tests/model_servering/server/llm/conftest.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+from kubernetes.dynamic import DynamicClient
+from kubernetes.dynamic.exceptions import NotFoundError
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.namespace import Namespace
+from ocp_resources.resource import ResourceEditor, get_client
+from ocp_resources.secret import Secret
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.service_mesh_member_roll import ServiceMeshMemberRoll
+from ocp_resources.serving_runtime import ServingRuntime
+from pytest_testconfig import config as py_config
+
+from tests.model_servering.server.llm.utils import base64_encode_str
+from utilities.constants import TRANSFORMERS_CACHE_ENV_VAR
+from utilities.inference_service import InferenceServiceForTests
+
+
+@pytest.fixture(scope="session")
+def admin_client() -> DynamicClient:
+    return get_client()
+
+
+@pytest.fixture(scope="class")
+def model_namespace(request, admin_client: DynamicClient) -> Namespace:
+    with Namespace(
+        client=admin_client,
+        name=request.param["name"],
+    ) as ns:
+        ns.wait_for_status(status=Namespace.Status.ACTIVE, timeout=120)
+        yield ns
+
+
+@pytest.fixture(scope="class")
+def patched_service_mesh_member_roll(admin_client: DynamicClient, model_namespace: Namespace) -> ServiceMeshMemberRoll:
+    smmr = ServiceMeshMemberRoll(client=admin_client, namespace="istio-system", name="default")
+
+    if not smmr.exists:
+        raise NotFoundError("istio ServiceMeshMemberRoll does not exist")
+
+    members = smmr.instance.spec.members
+
+    if members:
+        members.append(model_namespace.name)
+    else:
+        members = [model_namespace.name]
+
+    with ResourceEditor(patches={smmr: {"spec": {"members": members}}}):
+        yield smmr
+
+
+@pytest.fixture(scope="class")
+def storage_uri(request) -> str:
+    return f"s3://{py_config['model_s3_bucket_name']}/{request.param['model-dir']}/"
```

Comment:
"This should not be hardcoded. We can use any storage provider, such as S3 or Google Cloud Storage."

Reply:
Well arguably it's looking for `model_s3_bucket_name` so it makes sense to hardcode to S3, but in general yes, we should either have a generalized version of this fixture for any possible storage (not a good idea imho) or we can implement different fixtures for different protocols

Reply:
> we can implement different fixtures for different protocols

this is the right way. and, of course, the fixture name needs to indicate what is the storage type (e.g `aws_storage_api`